### PR TITLE
MM-25261: Add 'include_deleted' parameter to get all channels API req…

### PR DESCRIPTION
**Pull request by mkraft**
…uest.

#### Summary

Adds parameter to `include_deleted` channels via an API endpoint.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-25261

#### Dependent PR

https://github.com/mattermost/enterprise/pull/684

### CODE REVIEW

File: api4/channel_test.go [hunk 1]
```diff
@@ -925,6 +925,38 @@ func TestGetAllChannels(t *testing.T) {
...
+
+		channels, resp = client.GetAllChannels(0, 10000, "")
+		require.Nil(t, resp.Error)
+		beforeCount := len(*channels)
+
+		var firstChannel model.Channel
+		for _, item := range *channels {
+			firstChannel = item.Channel
+			break
+		}
```
Comment (fmunshi) [inline api4/channel_test.go:937]: why not just do `firstChannel := *channels[0].Channel`

Comment (mkraft) [reply api4/channel_test.go]: Meant to switch that, thanks.

**Comment by mkraft** [comment]
Docs: https://github.com/mattermost/mattermost-api-reference/pull/531

**Comment by srkgupta** [comment]
Added few issues in the dependent PR:
https://github.com/mattermost/mattermost-webapp/pull/5635

Waiting for it to be resolved.

**Pull request merged**